### PR TITLE
introduce randomness_config and jwk_consensus_config and deprecate the corresponding feature flags

### PR DESCRIPTION
**Pull request by zjma**
## Description

The on-chain configs of randomness components and JWK consensus should be standalone resources instead of feature flags.

`randomness_config.move`
```
    /// The configuration of the on-chain randomness feature.
    struct RandomnessConfig has drop, key, store {
        /// A config variant packed as an `Any`.
        /// Currently the variant type is one of the following.
        /// - `ConfigOff`
        /// - `ConfigV1`
        variant: Any,
    }

    /// A randomness config variant indicating the feature is disabled.
    struct ConfigOff has copy, drop, store {}

    /// A randomness config variant indicating the feature is enabled.
    struct ConfigV1 has copy, drop, store {
        /// Any validator subset should not be able to reconstruct randomness if `subset_power / total_power <= secrecy_threshold`,
        secrecy_threshold: FixedPoint64,
        /// Any validator subset should be able to reconstruct randomness if `subset_power / total_power > reconstruction_threshold`.
        reconstruction_threshold: FixedPoint64,
    }
```

`jwk_consensus_config.move`
```
    /// The configuration of the JWK consensus feature.
    struct JWKConsensusConfig has drop, key, store {
        /// A config variant packed as an `Any`.
        /// Currently the variant type is one of the following.
        /// - `ConfigOff`
        /// - `ConfigV1`
        variant: Any,
    }

    /// A JWK consensus config variant indicating JWK consensus should not run.
    struct ConfigOff has copy, drop, store {}

    struct OIDCProvider has copy, drop, store {
        name: String,
        config_url: String,
    }

    /// A JWK consensus config variant indicating JWK consensus should run to watch a given list of OIDC providers.
    struct ConfigV1 has copy, drop, store {
        oidc_providers: vector<OIDCProvider>,
    }
```

Regarding how to store a config object on chain:
- Storing the bcs bytes of a config object (like what's being done for `ConsensusConfig`) prevents us from inspecting config inside move (but that's what we want to do in `aptos_governance::reconfigure()`.
  Writing a native function to check certain fields should be avoided because of backward compatibility complexity in rust: even we never alter existing variants in the enum, someone may be able to "predict" the next variant struct and trigger execution divergence.
- Storing each field in its separate resource can work but will create too many functions in the module: a `set_x_for_next_epoch` function for every field `x`.
- Storing a single `Any` object avoids the issues above. IMO this is the best option.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [x] Refactoring
- [ ] Dependency update
- [ ] Documentation update

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [x] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
UT and smoke tests.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by danielxiangzl — COMMENTED** [review]
Now the new configs do not need to register in `ON_CHAIN_CONFIG_REGISTRY`? Is this removed? cc @zekun000

File: types/src/on_chain_config/randomness_config.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+#[derive(Deserialize, Serialize)]
+pub struct ConfigOff {}
+
+impl AsMoveAny for ConfigOff {
+    const MOVE_TYPE_NAME: &'static str = "0x1::randomness_config::ConfigOff";
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct ConfigV1 {}
```
[the remark this answers is not in the document]
Comment (zjma) [reply types/src/on_chain_config/randomness_config.rs:20]: is it actually:
```
V1 { secrecy_threshold: f64, reconstruction_threshold: f64,}, // fallback path only
V2 { secrecy_threshold: f64, reconstruction_threshold: f64, fast_path_secrecy_threshold: f64 } // fastpath + fallback path, to be added when landing fastpath on main
```
?

[the remark this answers is not in the document]
Comment (danielxiangzl) [reply types/src/on_chain_config/randomness_config.rs:20]: I don't have a strong opinion. Maybe what you suggested is better.

File: consensus/src/epoch_manager.rs [hunk 2]
```diff
@@ -1032,11 +1038,12 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
                 epoch_state,
                 consensus_config,
                 execution_config,
+                onchain_randomness_config,
+                jwk_consensus_config,
```
Comment (zekun000) [inline consensus/src/epoch_manager.rs:1042]: why it needs to understand jwk config?

File: testsuite/smoke-test/src/validator_txns.rs [hunk 3]
```diff
@@ -19,13 +19,9 @@ async fn dummy_validator_txns() {
...
         .with_init_genesis_config(Arc::new(move |conf| {
-            // start with vtxn disabled.
-            conf.consensus_config.disable_validator_txns();
-
-            // start with dkg enabled.
-            let mut features = Features::default();
-            features.enable(FeatureFlag::RECONFIGURE_WITH_DKG);
-            conf.initial_features_override = Some(features);
+            // start with randomness enabled.
+            conf.consensus_config.enable_validator_txns();
```
Comment (zekun000) [inline testsuite/smoke-test/src/validator_txns.rs:23]: looks like the test changed? from disable vtxn to enable?

**Review by zekun000 — APPROVED** [review]

**Review by danielxiangzl — APPROVED** [review]

File: testsuite/smoke-test/src/validator_txns.rs (hunk 3 above)
Comment (zjma) [reply testsuite/smoke-test/src/validator_txns.rs:23]: yes, that's to allow dummy dkg txns through proposal and get executed.

before we were using `DummyTransaction` which does not exist any more.

File: consensus/src/epoch_manager.rs (hunk 2 above)
Comment (zjma) [reply consensus/src/epoch_manager.rs:1048]: ircc that's for round manager needs to know whether the jwk txn in proposal is expected

**Pull request merged**